app/models/concerns/entity_access.rb

```diff
@@ -49,26 +82,76 @@ def editable_access_control
     ).first_or_create
   end
 
+  ##
+  # Finds or creates a "system" +Collection+ dedicated to this entity.
+  # The collection will only contain one group-viewable entity (this one). It
+  # also updates the collection's name to match the entity's +name+.
+  # Note that updating the name may cause an extra query, and in the future we may wan to
+  # refactor that into an after save hook or similar.
+  #
+  # @return [Collection]
+  #
   def system_collection
     @system_collection ||= begin
-      collection = Collection.where(system: ['Entities'], name: name, collection_type: collection_type).first_or_create
+      collection = Collection.system.where(source: self).first_or_create
       collection.set_viewables(entity_relation_type => [id])
+      collection.update(name: name) # ensure the collection name still matches
```

Comment:
(same throughout this file)
```suggestion
      collection.update!(name: name) # ensure the collection name still matches
```